org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/TaskManager.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2015 Contributor. All rights reserved.
+ */
+package org.scalaide.core.internal.builder
+
+import scala.reflect.internal.util.Position
+import scala.reflect.internal.util.SourceFile
+
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IMarker
+import org.eclipse.jdt.core.IJavaModelMarker
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.jdt.core.compiler.IProblem
+import org.eclipse.jdt.internal.core.builder.JavaBuilder
+import org.scalaide.core.IScalaProject
+import org.scalaide.core.SdtConstants
+import org.scalaide.core.internal.jdt.model.ScalaSourceFile
+import org.scalaide.core.resources.EclipseResource
+
+import scalariform.lexer.ScalaLexer
+
+object TaskManager {
+
+  private case class Comment(msg: String, pos: Position)
+
+  /**
+   * Updates all tasks (`TODO`s and `FIXME`s) for given project in given files.
+   */
+  def updateTasks(project: IScalaProject, files: Set[IFile]): Unit = {
+    val taskScanner = new TaskScanner(project)
+
+    for {
+      iFile <- files
+      Comment(msg, pos) <- extractComments(iFile)
+      if pos.isDefined
+      TaskScanner.Task(tag, msg, priority, taskPos) <- taskScanner.extractTasks(msg, pos)
+      if taskPos.isDefined
+    } {
+      val start = taskPos.start
+      val length = taskPos.end - start
+      taskPos.source.file match {
+        case EclipseResource(i: IFile) =>
+          registerTask(i, tag, msg, priority, start, length, taskPos.line)
```

Comment:
wrapRegisterTask in method that takes 2 arguments i (btw rename i to some more meaningfull)

question to all: Can't we use iFile from above?